packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -1,889 +1,275 @@
+    useQueryContext()
+  const { columnsToShowInTable, NoContentPlaceholder } =
+    useQueryVisualizationContext()
+  const synapseTableContext = useMemo(() => ({ columnLinks }), [columnLinks])
 
-  // instance variables
-  resizer: any
-  tableElement: HTMLTableElement | null | undefined = undefined
+  const { columnModels = [] } = data ?? {}
 
-  componentWillUnmount() {
-    this.disableResize()
-  }
+  const isLoggedIn = !!useSynapseContext().accessToken
 
-  componentDidMount() {
-    this.getEntityHeadersInData(true)
-    this.enableResize()
-  }
+  const [isExportTableDownloadOpen, setIsExportTableDownloadOpen] =
+    useState(false)
 
-  shouldComponentUpdate(
-    nextProps: SynapseTableProps,
-    nextState: Readonly<SynapseTableState>,
-  ): boolean {
-    // ignore isFetching state variables when checking for change in state
-    const {
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      isFetchingEntityHeaders: oldIsFetchingEntityHeaders,
-      ...oldState
-    } = this.state
-    const {
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      isFetchingEntityHeaders: newIsFetchingEntityHeaders,
-      ...newState
-    } = nextState
-    const isPropsChange = !eq(this.props, nextProps)
-    const isStateChange = !eq(oldState, newState)
-    const shouldComponentUpdate = isPropsChange || isStateChange
-    if (shouldComponentUpdate) {
-      this.disableResize()
-    }
-    return shouldComponentUpdate
-  }
-  componentDidUpdate(
-    prevProps: SynapseTableProps,
-    prevState: Readonly<SynapseTableState>,
-  ) {
-    // PORTALS-2081: if the data changed, then get the new entity headers
-    if (!eq(prevProps.queryContext.data, this.props.queryContext.data)) {
-      this.getEntityHeadersInData(false)
-    } else if (
-      prevProps.queryContext.entity &&
-      this.props.queryContext.entity &&
-      !eq(
-        isEntityView(prevProps.queryContext.entity),
-        isEntityView(this.props.queryContext.entity),
-      )
-    ) {
-      // if we determined that this is an entity view, force refresh the entity headers (for Views we need to get the rowIds!)
-      this.getEntityHeadersInData(true)
-    }
-    this.enableResize()
-  }
-
-  enableResize() {
-    if (!this.resizer) {
-      // only create the resizer if the tableElement is in the DOM and there's more than one select column in the data.
-      if (this.tableElement && this.getLengthOfPropsData() > 1) {
-        this.resizer = new ColumnResizer(this.tableElement, RESIZER_OPTIONS)
-      }
-    } else {
-      // TODO: use event driven solution instead of timeout.
-      // We need to give SynapseTableCell time to render (which will change the column size).
-      // Alternatively, we may be able to use SizeMe (or withSize() on Cell component) to respond to width change.
-      setTimeout(() => {
-        this.resizer.reset(RESIZER_OPTIONS)
-      }, 1000)
-    }
-  }
+  const { sort, setSort } = useTableSort()
```

Comment:
Custom hook to map between our table query sorting logic and `@tanstack/react-table`'s sorting logic (just for keeping track of which columns are sorted, no client-side data sorting happening here)